rest/src/main/java/fi/otavanopisto/pyramus/rest/controller/StudentController.java

```diff
@@ -497,18 +582,63 @@ public List<CourseActivity> listCourseActivities(List<CourseStudent> courseStude
           : Collections.emptyList();
       for (CreditLink creditLink : linkedTransferCredits) {
         TransferCredit transferCredit = (TransferCredit) creditLink.getCredit();
+        
+        // Basic info
+        
         CourseActivity courseActivity = new CourseActivity();
-        courseActivity.setCurriculumIds(Collections.emptyList());
         courseActivity.setCourseName(transferCredit.getCourseName());
-        courseActivity.setGrade(transferCredit.getGrade().getName());
-        courseActivity.setPassingGrade(transferCredit.getGrade().getPassingGrade());
-        courseActivity.setActivityDate(transferCredit.getDate());
-        courseActivity.setState(CourseActivityState.TRANSFERRED);
+        
+        // Curriculum
+
+        if (transferCredit.getCurriculum() != null) {
+          CourseActivityCurriculum cac = new CourseActivityCurriculum();
+          cac.setId(transferCredit.getCurriculum().getId());
+          cac.setName(transferCredit.getCurriculum().getName());
+          courseActivity.setCurriculums(Arrays.asList(cac));
+        }
+        else {
+          courseActivity.setCurriculums(Collections.emptyList());
+        }
+        
+        // Subject
+        
+        CourseActivitySubject subject = new CourseActivitySubject();
+        if (transferCredit.getCourseLength() != null) {
+          subject.setCourseLength(transferCredit.getCourseLength().getUnits());
+          subject.setCourseLengthSymbol(transferCredit.getCourseLength().getUnit().getSymbol());
+        }
+        subject.setCourseNumber(transferCredit.getCourseNumber());
+        if (transferCredit.getSubject() != null) {
+          subject.setSubjectCode(transferCredit.getSubject().getCode());
+          subject.setSubjectName(transferCredit.getSubject().getName());
+        }
+        courseActivity.setSubjects(Arrays.asList(subject));
+        
+        // Assessment
+
+        CourseActivityAssessment assessment = new CourseActivityAssessment();
+        if (transferCredit.getGrade() != null) {
+          assessment.setGrade(transferCredit.getGrade().getName());
+          assessment.setPassingGrade(transferCredit.getGrade().getPassingGrade());
+          assessment.setGradeDate(transferCredit.getDate());
+        }
+        assessment.setDate(transferCredit.getDate());
+        assessment.setState(CourseActivityState.TRANSFERRED);
+        courseActivity.setAssessments(Arrays.asList(assessment));
+
         courseActivities.add(courseActivity);
       }
     }
     
-    return courseActivities;
+    // Basic info about the student complemented with the activity list above
+    
+    CourseActivityInfo courseActivityInfo = new CourseActivityInfo();
+    courseActivityInfo.setLineName(student.getStudyProgramme().getName());
+    courseActivityInfo.setLineCategory(student.getStudyProgramme().getCategory().getName());
```

Comment:
Is studyprogramme and its category always present?

Reply:
Every student has a study programme and every study programme has to have a category (although for study programmes, Pyramus UI shows both organization and category fields as non-mandatory but trying to save a new study programme without them leads to an Internal Server Error dialog ^^;)